Assembler.Base/FramesAssembler.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Collections.Generic;
+using Assembler.Core;
+using Assembler.Core.Entities;
+using Assembler.Core.Enums;
+using Microsoft.Extensions.Logging;
+
+namespace Assembler.Base
+{
+    public class FramesAssembler<TFrame> : IAssembler<TFrame>
+        where TFrame : BaseFrame
+    {
+        private readonly IResolver<AssemblingPosition, IFrameHandler<TFrame>> _typeToHandlerResolver;
+        private readonly ILogger _logger;
+
+        public FramesAssembler(IResolver<AssemblingPosition, IFrameHandler<TFrame>> typeToHandlerResolver,
+            ILoggerFactory loggerFactory)
+        {
+            _typeToHandlerResolver = typeToHandlerResolver;
+            _logger = loggerFactory.CreateLogger(GetType());
+        }
+
+        public void Assemble(TFrame frame)
+        {
+            IFrameHandler<TFrame> handler = ResolveHandler(frame);
+
+            _logger.LogDebug($"Frame [{frame.Guid}] is being sent to the handler.");
+
+            HandleFrame(frame, handler);
+        }
+
+        private IFrameHandler<TFrame> ResolveHandler(TFrame frame)
+        {
+            try
+            {
+                return _typeToHandlerResolver.Resolve(frame.AssemblingPosition);
+            }
+            catch (Exception e)
+            {
+                _logger.LogWarning($"No matching resolver was found for the message [{frame.Guid}]," +
```

Comment:
1. No matching *handler*
2. Do you think thats Warn or Error? Im not sure

Reply:
I think that because the scope is a single frame we could get away with warn